sound/soc/sof/ipc4.c

```diff
@@ -685,7 +685,11 @@ static void sof_ipc4_rx_msg(struct snd_sof_dev *sdev)
 			return;
 
 		ipc4_msg->data_size = data_size;
-		snd_sof_ipc_msg_data(sdev, NULL, ipc4_msg->data_ptr, ipc4_msg->data_size);
+		err = snd_sof_ipc_msg_data(sdev, NULL, ipc4_msg->data_ptr, ipc4_msg->data_size);
+		if (err < 0) {
+			dev_err(sdev->dev, "failed to read IPC notification data: %d\n", err);
+			return err;
```

Comment:
you still need to do the cleanup for the allocated buffer...

Reply:
> you still need to do the cleanup for the allocated buffer...

fixed